webservice/webservice.py

```diff
@@ -25,9 +27,33 @@
 # how-to-write-docstring-for-url-parameters
 
 
+def authorize_with_dashboard(f):
+    @functools.wraps(f)
+    def check_auth(*args, **kwargs):
+        headers = {}
+        # look at header for what we need
+        if 'Authorization' in request.headers:
+            headers['Authorization'] = request.headers['Authorization']
+        # add cookie to request for auth
+        if 'Cookie' in request.headers:
+            headers['Cookie'] = request.headers['Cookie']
+        # set the X-Forwarded-For and User-agent to prevent session protection on dashboard from causing problems
+        headers['X-Forwarded-For'] = request.headers.get('X-Forwarded-For', request.remote_addr)
+        if 'User-Agent' in request.headers:
+            headers['User-Agent'] = request.headers['User-Agent']
+        # make call to dashboard
+        res = requests.get(request.url_root + 'authorization', headers=headers)
+        if res.status_code == 204:
```

Comment:
Optional: Although endpoint returns 204, it could be safer to check for 2xx. Sort of torn on this one... Your call.

Reply:
I'm going to leave it unless you convince me otherwise. I have no problem changing though.

Reply:
Sounds good, if I can't can come up with a convincing argument, then leave it as is. :)